[CI] MatchFunctionIT testMatchWithLookupJoinOnMatch failing

**Build Scans:**
- [elasticsearch-intake #30521 / part3](https://gradle-enterprise.elastic.co/s/ffvnns2jwz3o2)
- [elasticsearch-pull-request #100743 / part-3](https://gradle-enterprise.elastic.co/s/pzywpvpyudwbk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.plugin.MatchFunctionIT.testMatchWithLookupJoinOnMatch" -Dtests.seed=48C6AEDAE8B8A530 -Dtests.locale=ksb -Dtests.timezone=Europe/Sarajevo -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.plugin.MatchFunctionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testMatchWithLookupJoinOnMatch'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <[[This is a brown fox, 1, 1, This is a brown fox], [The quick brown fox jumps over the lazy dog, 6, 6, The quick brown fox jumps over the lazy dog]]>
     but: was <[[The quick brown fox jumps over the lazy dog, 6, 6, The quick brown fox jumps over the lazy dog], [This is a brown fox, 1, 1, This is a brown fox]]>
```

**Issue Reasons:**
- [main] 2 failures in test testMatchWithLookupJoinOnMatch (1.5% fail rate in 135 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -504,18 +504,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeMetricsIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/137071
-- class: org.elasticsearch.xpack.esql.plugin.MatchFunctionIT
-  method: testMatchWithLookupJoinOnMatch
-  issue: https://github.com/elastic/elasticsearch/issues/137137
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:lookup-join-expression.LookupJoinExpressionWithTerm}
-  issue: https://github.com/elastic/elasticsearch/issues/137157
-- class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecIT
-  method: test {csv-spec:lookup-join-expression.LookupJoinExpressionWithTerm}
-  issue: https://github.com/elastic/elasticsearch/issues/137157
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeForkIT
-  method: test {csv-spec:lookup-join-expression.LookupJoinExpressionWithTerm}
-  issue: https://github.com/elastic/elasticsearch/issues/137160
 - class: org.elasticsearch.cluster.routing.allocation.decider.WriteLoadConstraintDeciderIT
   method: testCanRemainNotPreferredIsIgnoredWhenAllOtherNodesReturnNotPreferred
   issue: https://github.com/elastic/elasticsearch/issues/137162

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join-expression.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join-expression.csv-spec
@@ -957,6 +957,7 @@ id_left:integer | name_left:keyword | other1_from_first_join:keyword | other1:ke
 lookupJoinExpressionWithTerm
 required_capability: join_lookup_v12
 required_capability: lookup_join_with_full_text_function
+required_capability: term_function
 
 FROM multi_column_joinable
 | RENAME id_int AS id_left, is_active_bool AS is_active_left
@@ -1016,6 +1017,7 @@ id_left:integer | name_str:keyword | extra1:keyword | other1:keyword | other2:in
 lookupJoinExpressionWithMultiMatch
 required_capability: join_lookup_v12
 required_capability: lookup_join_with_full_text_function
+required_capability: multi_match_function
 
 FROM multi_column_joinable
 | RENAME id_int AS id_left, is_active_bool AS is_active_left, name_str AS name_str_left

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/MatchFunctionIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/MatchFunctionIT.java
@@ -319,6 +319,7 @@ public class MatchFunctionIT extends AbstractEsqlIntegTestCase {
             | rename id as id_left
             | LOOKUP JOIN test_lookup ON id_left == id and MATCH(lookup_content, "fox")
             | WHERE id > 0
+            | SORT id, id_left, content, lookup_content
             """;
         try (var resp = run(query)) {
             assertColumnNames(resp.columns(), List.of("content", "id_left", "id", "lookup_content"));

